File: zig/examples/basic.zig
const std = @import("std");
const SensitiveString = @import("sensitive_string").SensitiveString;

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    std.debug.print("=== Zig SensitiveString Example ===\n\n", .{});

    // Create a sensitive string
    const password = try SensitiveString.init(allocator, "my-secret-password");
    defer password.deinit();

    // Get the hash string
    const hash1 = try password.hashHex(allocator);
    defer allocator.free(hash1);
    std.debug.print("Format: {s}\n", .{hash1});

    // Intentional access to plaintext
    std.debug.print("\n=== Intentional Access ===\n", .{});
    std.debug.print("Plaintext (via .value): {s}\n", .{password.value});
    std.debug.print("Plaintext (via getValue()): {s}\n", .{password.getValue()});

    // Multiple sensitive strings
    const api_key = try SensitiveString.init(allocator, "sk-1234567890abcdef");
    defer api_key.deinit();

    const hash2 = try api_key.hashHex(allocator);
    defer allocator.free(hash2);
    std.debug.print("\nAPI Key: {s}\n", .{hash2});

    var buffer: [1024]u8 = undefined;
    var file_writer = std.fs.File.stdout().writer(&buffer);
    const writer_ptr = &file_writer.interface;
    try std.json.Stringify.value(
        password,
        .{ .whitespace = .indent_2 },
        writer_ptr,
    );
    try file_writer.interface.flush();
    

    std.debug.print("\n✅ Notice: All output shows SHA256 hashes, not plaintext!\n", .{});
}

File: rust/examples/basic.rs
//! Basic usage example of SensitiveString
//!
//! Run with: cargo run --example basic

use sensitive_string::SensitiveString;

fn main() {
    println!("=== Basic SensitiveString Usage ===\n");

    // Create a sensitive string
    let password = SensitiveString::new("my-secret-password".to_string());

    // Display formatting - shows hash
    println!("Display format: {}", password);
    println!("Debug format: {:?}", password);

    // Can also use .into() for convenience
    let api_key: SensitiveString = "sk-1234567890abcdef".into();
    println!("\nAPI Key: {}", api_key);

    // Intentional access when you need the plaintext
    println!("\n=== Intentional Access ===");
    println!("Plaintext (via get_value): {}", password.get_value());
    println!("Plaintext (via value): {}", password.value());

    // Utility methods
    println!("\n=== Utility Methods ===");
    println!("Length: {}", password.len());
    println!("Is empty: {}", password.is_empty());

    // Equality
    let password2 = SensitiveString::new("my-secret-password".to_string());
    println!("Equal to copy: {}", password == password2);

    // Clone
    let cloned = password.clone();
    println!("Equal to clone: {}", password == cloned);

    // String concatenation - shows hash
    println!("\n=== String Operations ===");
    let message = format!("Your password is: {}", password);
    println!("{}", message);
}

